src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/CQLSAIIndex.java

```diff
@@ -102,11 +107,14 @@ static boolean indexClassIsSai(String className) {
    * <p>The Reg Exp below will match:
    *
    * <ul>
-   *   <li>"monkeys": group 1 - "monkeys", group 2 - null
-   *   <li>"values(monkeys)": group 1 - "values" group 2 - "monkeys"
+   *   <li>'monkeys': group 1 - 'monkeys', group 2 - null
+   *   <li>'values(monkeys)': group 1 - 'values' group 2 - 'monkeys'
+   *   <li>'"Monkeys": group 1 - '"Monkeys"', group 2 - null
+   *   <li>'values("Monkeys")': group 1 - 'values' group 2 - '"monkeys"'
    * </ul>
    */
-  private static Pattern INDEX_TARGET_PATTERN = Pattern.compile("^(\\w+)?(?:\\((\\w+)\\))?$");
+  private static Pattern INDEX_TARGET_PATTERN =
+      Pattern.compile("^(\"?\\w+\"?)?(?:\\((\"?\\w+\"?)\\))?$");
```

Comment:
omg. This is... not easy to decipher. I don't know if anything can be done to help since regexps get pretty complicated for nested cases.

Reply:
Agree, let me write some tests at least.